fairroot/online/source/FairLmdSource.cxx

```diff
@@ -54,6 +54,7 @@ FairLmdSource::FairLmdSource(const FairLmdSource& source)
 
 FairLmdSource::~FairLmdSource()
 {
+    CloseLmd();
```

Comment:
This step should be executed by its owner.

Reply:
Well, no.

I don't think, the "owner" should be resposible for calling Close. It could be considered "being a good citizen", but it should not be strictly needed.

```cpp
FairLmdSource mysource;
mysource.Open(…);
if (true)
  throw std::runtime_exception("Something else went wrong");
mysource.Close();
```

This should not leak resources! (See "RAII printciple").

So yes, This is some overhead for the developer of a FairSource/Sink. But that's how C++ is like.